### PR TITLE
Sharing: add social previews

### PR DESCRIPTION
**Pull request by gwwar**
Most of the code is copied from our SEO preview panel, with a bit of cleanup. For the purposes of this meetup, I pulled this out so we can diverge and update styling safely. If there are any common patterns we can refactor this later to use in more common components.

screen shot 2017-03-21 at 12 37 46 pm

### Testing Instructions

- Navigate to http://calypso.localhost:3000/devdocs/blocks/sharing-preview-pane
- You should be able to preview a post. Note that posts that have images that are too small will not get an image preview.

### CODE REVIEW

File: client/blocks/sharing-preview-pane/docs/example.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	return {
+		siteId,
+		postId
+	};
+} )( SharingPreviewPaneExample );
+
+ConnectedSharingPreviewPaneExample.displayName = 'SharingPreviewPane';
+
+export default ConnectedSharingPreviewPaneExample;
```
Comment (retrofox) [inline client/blocks/sharing-preview-pane/docs/example.jsx]: new line here

File: client/blocks/sharing-preview-pane/index.jsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	const seoTitle = getSeoTitle( state, 'posts', { site, post } );
+
+	return {
+		site,
+		post,
+		seoTitle
+	};
+};
+
+const mapDispatchToProps = {};
```
Comment (retrofox) [inline client/blocks/sharing-preview-pane/index.jsx]: is it needed

Comment (gwwar) [inline client/blocks/sharing-preview-pane/index.jsx]: removed in 8873b73

File: client/blocks/sharing-preview-pane/docs/example.jsx (hunk 1 above)
Comment (gwwar) [inline client/blocks/sharing-preview-pane/docs/example.jsx]: added in 8873b73

File: client/blocks/sharing-preview-pane/index.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+							{ translate( 'Social Previews' ) }
+						</h1>
+						<p className="sharing-preview-pane__description">
+							{ translate(
+								'This is how your post will appear ' +
+								'when people view or share it on any of ' +
+								'the networks below' ) }
+						</p>
+					</div>
+					<VerticalMenu onClick={ this.selectPreview }>
```
Comment (lamosty) [inline client/blocks/sharing-preview-pane/index.jsx]: Can we pass in a `className` so we don't have to do the following nesting?

selection_014

**Comment by lamosty** [comment]
Also, what about merging this into `master` so the `add/publicize-scheduling` branch is not extremely huge? These should be reusable components anyways.

**Comment by lamosty** [comment]
Also, what about merging this into `master` so the `add/publicize-scheduling` branch is not extremely huge? These should be reusable components anyways.

File: client/blocks/sharing-preview-pane/index.jsx (hunk 3 above)
Comment (gwwar) [inline client/blocks/sharing-preview-pane/index.jsx]: I can follow up on this one :D in another PR

**Comment by gwwar** [comment]
> Also, what about merging this into master so the add/publicize-scheduling branch is not extremely huge?

When it's in a good state, I'll try to pick out the commits from the feature branch, or make a new PR

**Comment by retrofox** [comment]
> > Also, what about merging this into master so the add/publicize-scheduling branch is not extremely huge?

> When it's in a good state, I'll try to pick out the commits from the feature branch, or make a new PR

I think we should merge those changes in new PRs when it's possible.

**Pull request merged**